src/commands/selection.rs

```diff
@@ -100,6 +115,81 @@ fn copy_to_clipboard(app: &mut Application) -> Result {
     Ok(())
 }
 
+/// Get the selected range from an application in a selection mode. *Requires*
+/// that the application has a buffer and is in mode Select, SelectLine, or
+/// Search.
+fn range_from(app: &mut Application) -> std::result::Result<Range, Error> {
+    let buffer = app.workspace.current_buffer();
+    let buffer = buffer.unwrap();
+
+    Ok(match app.mode {
+        Mode::Select(ref select_mode) => {
+            Range::new(*buffer.cursor.clone(), select_mode.anchor)
+        }
+        Mode::SelectLine(ref select_line_mode) => {
+            select_line_mode.to_range(&*buffer.cursor)
+        }
+        Mode::Search(ref mode) => {
+            mode.results
+            .as_ref()
+            .and_then(|r| r.selection())
+            .ok_or("Cannot get selection outside of select mode.")
+            .unwrap()
+            .clone()
+        }
+        _ => bail!("Cannot get selection outside of select mode."),
+    })
+}
+
+/// Wrap a string at a given maximum length (generally 80 characters). If the
+/// line begins with a comment (matches potential_prefix), the text is wrapped
+/// around it.
+fn justify_string(text: &str, max_len: usize, potential_prefix: &str) -> String {
```

Comment:
I think it makes more sense to move the default value handling for a prefix into this function, and the `Option` type makes the "potential" aspect of this argument self-describing.

```suggestion
fn justify_string(text: &str, max_len: usize, prefix: Option<&str>) -> String {
```